### ISSUE TITLE
Sample weights not used in accuracy computation

### ISSUE THREAD

**Issue by danijel3** [fchollet/keras#1642]
Judging by the docs, this will probably get added one day, but I wanted to add some code suggestions, to possibly speed things up.

The loss function seems to take sample weights into account everywhere, by using the "weighted_loss" function in the code, but accuracy is the same, regardless if you use sample weights or not.

I made this simple method to compute the accuracy with sampled weights using numpy, but I guess it shouldn't be too hard to rewrite it into Keras/Theano:

```
def accuracy(model,inp,out,weigths):
    p=model.predict(inp)
    cnt=np.sum(weigths)    
    err=np.sum(np.not_equal(np.argmax(p,axis=-1)*weigths,np.argmax(out,axis=-1)*weigths))
    acc=1.0-(err/cnt)
    return acc
```

It actually works by computing the error rate instead of accuracy and then subtracting that from 1. It's easier to count errors, because then you don't have to consider the zeroed out samples in any special way. 

Oh and BTW, for categorical classification, this only works with weights that are 0 or 1. Modifying classes by other weight values obviously don't make any sense.

Finally, can I have one question: what's the difference between Masking/Sample weights? Can you use them both? Can you use only one of them? Is the reason for using Masking only to save time (cause I don't notice much difference with it)? Can Masking be used in the first layer now?

**Comment by fchollet** [comment]
"Weighting" in this context means "loss weighting", which the documentation is pretty explicit about. As a result, it only applies to the loss. If you need to monitor custom metrics that require weighting, you can implement your own.

> Finally, can I have one question: what's the difference between Masking/Sample weights?

"Masking" means masking specific timesteps in an input sequence. Sample weighting consist in weighting the impact of specific output samples in the loss function.

**Comment by maxbry** [comment]
I don't think this is very intuitive. If weighting actually only means loss weighting, then it should be called that. Can we get something like a true/false flag to use the weighting also for the accuracy?

**Comment by iskandr** [comment]
I also found this surprising: accuracy decreasing during training corresponded to an *increase* in weighted accuracy.

**Comment by tstandley** [comment]
I agree. Metrics calculated without weights seems wrong. If for some of my data, I don't have certain outputs, how can I tell how well it is doing on the rest of the data.

**Comment by utkrist** [comment]
Is there a way to pass the weight tensor to the custom accuracy metric? The function only seems to have two arguments y_true and y_pred.

**Comment by dickreuter** [comment]
I'm having the same problem. What if the sample weights are not symmetric (i.e. payoff when correctly classified, and loss when incorrectly classified). Is there a way to write a loss function and metric that would consider that?

**Comment by ftarlao** [comment]
I'm having the same problem. I need an accuracy metric that takes into account classes weights
